repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/classloader/ClassLoaderTest.groovy

```diff
@@ -92,7 +82,8 @@ class ClassLoaderTest extends ReposeValveTest {
         repose.start(killOthersBeforeStarting: false,
                 waitOnJmxAfterStarting: false)
 
-        repose.waitForNon500FromUrl(url)
+        reposeLogSearch.awaitByString("Repose ready", 1, 30)
```

Comment:
All good observations. I would very much so like to fix up our wait methods, however, my immediate concern with this PR is that we are adding yet place where we wait for Repose to start.

So if in the future we change the "Repose ready" message, this gives us another place to update. Or if we come up with a foolproof solution, this is a place where we might miss updating.

As for how to handle that, I'm not sure. I'd love to clean up and consolidate (at least to a single class) the waiting methods that we have.